packages/block-library/src/navigation/index.php

```diff
@@ -121,7 +132,8 @@ function build_navigation_html( $block, $colors, $font_sizes ) {
 		? sprintf( ' style="%s"', esc_attr( $colors['inline_styles'] ) . esc_attr( $font_sizes['inline_styles'] ) )
 		: '';
 
-	foreach ( (array) $block['innerBlocks'] as $key => $block ) {
+	// Only include link blocks that have a label set.
+	foreach ( (array) array_filter( $block['innerBlocks'], 'has_navigation_label' ) as $key => $block ) {
```

Comment:
The `(array)` casting should probably be kept in place for `$block['innerBlocks']`. `array_filter()` will always return an array. Same below